### PR TITLE
Annotate System.Collections.NonGeneric for nullable

### PR DESCRIPTION
**Pull request by buyaa-n**
Contributes to #40623
cc: @dotnet/nullablefc

### CODE REVIEW

File: src/System.Collections.NonGeneric/ref/System.Collections.NonGeneric.cs [hunk 1]
```diff
@@ -36,25 +35,25 @@ public abstract partial class CollectionBase : System.Collections.ICollection, S
         object System.Collections.ICollection.SyncRoot { get { throw null; } }
         bool System.Collections.IList.IsFixedSize { get { throw null; } }
         bool System.Collections.IList.IsReadOnly { get { throw null; } }
-        object System.Collections.IList.this[int index] { get { throw null; } set { } }
+        object? System.Collections.IList.this[int index] { get { throw null; } set { } }
```
Comment (krwq) [inline src/System.Collections.NonGeneric/ref/System.Collections.NonGeneric.cs]: should InnerList  return nullable?

File: src/System.Collections.NonGeneric/ref/System.Collections.NonGeneric.cs [hunk 2]
```diff
@@ -66,23 +65,23 @@ public abstract partial class DictionaryBase : System.Collections.ICollection, S
         object System.Collections.ICollection.SyncRoot { get { throw null; } }
         bool System.Collections.IDictionary.IsFixedSize { get { throw null; } }
         bool System.Collections.IDictionary.IsReadOnly { get { throw null; } }
-        object System.Collections.IDictionary.this[object key] { get { throw null; } set { } }
+        object? System.Collections.IDictionary.this[object key] { get { throw null; } set { } }
```
Comment (krwq) [inline src/System.Collections.NonGeneric/ref/System.Collections.NonGeneric.cs]: should InnerHashtable be nullable?

File: src/System.Collections.NonGeneric/ref/System.Collections.NonGeneric.cs [hunk 3]
```diff
@@ -100,12 +99,12 @@ public partial class Queue : System.Collections.ICollection, System.Collections.
         public virtual object Clone() { throw null; }
         public virtual bool Contains(object obj) { throw null; }
         public virtual void CopyTo(System.Array array, int index) { }
-        public virtual object Dequeue() { throw null; }
-        public virtual void Enqueue(object obj) { }
+        public virtual object? Dequeue() { throw null; }
```
Comment (krwq) [inline src/System.Collections.NonGeneric/ref/System.Collections.NonGeneric.cs]: should Conatins take nullable?

File: src/System.Collections.NonGeneric/ref/System.Collections.NonGeneric.csproj [hunk 4]
```diff
@@ -1,6 +1,8 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <Configurations>netcoreapp-Debug;netcoreapp-Release;uap-Debug;uap-Release</Configurations>
+    <DefineConstants>$(DefineConstants);INTERNAL_NULLABLE_ATTRIBUTES</DefineConstants>
```
Comment (krwq) [inline src/System.Collections.NonGeneric/ref/System.Collections.NonGeneric.csproj]: where is this used?

Comment (krwq) [reply src/System.Collections.NonGeneric/ref/System.Collections.NonGeneric.csproj]: (I can see attributes being used for implementation assembly but not the ref)

File: src/System.Collections.NonGeneric/ref/System.Collections.NonGeneric.cs (hunk 1 above)
Comment (buyaa-n) [reply src/System.Collections.NonGeneric/ref/System.Collections.NonGeneric.cs]: yes, that is ArrayList and indexer return nullable

File: src/System.Collections.NonGeneric/ref/System.Collections.NonGeneric.cs (hunk 2 above)
Comment (buyaa-n) [reply src/System.Collections.NonGeneric/ref/System.Collections.NonGeneric.cs]: yes

File: src/System.Collections.NonGeneric/ref/System.Collections.NonGeneric.cs (hunk 3 above)
Comment (buyaa-n) [reply src/System.Collections.NonGeneric/ref/System.Collections.NonGeneric.cs]: No as that parameter is key which shouldn't be nullable

File: src/System.Collections.NonGeneric/ref/System.Collections.NonGeneric.csproj (hunk 4 above)
Comment (buyaa-n) [reply src/System.Collections.NonGeneric/ref/System.Collections.NonGeneric.csproj]: for ref assembly

Comment (buyaa-n) [reply src/System.Collections.NonGeneric/ref/System.Collections.NonGeneric.csproj]: > (I can see attributes being included for implementation assembly but not the ref)
oh I see what you mean, will remove

**Pull request merged**